File: src/front-end-nextjs/hob-dashboard/lib/types/sale.ts
export interface Sale {
  saleId: string;
  orderId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  createdAt: string;
  updatedAt?: string;
  order?: {
    orderDate: string;
    customerName: string;
    status: string;
    salesCount: number;
  };
  // Legacy field for backward compatibility
  orderDetails?: {
    orderDate: string;
    customerName: string;
    status: string;
  };
}

export interface CreateSaleRequest {
  orderId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
}

export interface UpdateSaleRequest {
  quantity: number;
  unitPrice: number;
}

File: src/HOB.Data/Entities/Sale.cs
namespace HOB.Data.Entities;

public class Sale
{
    public Guid SaleId { get; set; }
    public Guid OrderId { get; set; }
    public string ProductName { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal TotalPrice { get; set; }
    public DateTime CreatedAt { get; set; }

    // Navigation property
    public virtual Order Order { get; set; } = null!;
}
